### ISSUE TITLE
com.hazelcast.jet.server.JetCommandLineTest.testTargetsAfterCommandTakesPrecedence

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast-jet#2526]
_master_ (commit 73fe09d5cac40178068a48d23bfb0f8ee61d0ec8)

This issue can be caused by enormous hiccups during the test, see below.

Failed on Sonar nightly build: http://jenkins.hazelcast.com/job/jet-oss-master-sonar-nightly/618/testReport/junit/com.hazelcast.jet.server/JetCommandLineTest/testTargetsAfterCommandTakesPrecedence/

Stacktrace:
```
java.lang.NullPointerException
	at com.hazelcast.jet.server.JetCommandLineTest.testTargetsCommandCluster(JetCommandLineTest.java:564)
	at com.hazelcast.jet.server.JetCommandLineTest.testTargetsCommand(JetCommandLineTest.java:547)
	at com.hazelcast.jet.server.JetCommandLineTest.testTargetsAfterCommandTakesPrecedence(JetCommandLineTest.java:528)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output:
```
Finished Running Test: test_submit_clientShutdownWhenDone in 426.013 seconds.
Hiccups measured while running test 'testTargetsAfterCommandTakesPrecedence(com.hazelcast.jet.server.JetCommandLineTest):'
13:36:50, accumulated pauses: 361 ms, max pause: 99 ms, pauses over 1000 ms: 0
13:36:55, accumulated pauses: 3749 ms, max pause: 3743 ms, pauses over 1000 ms: 1
13:37:00, accumulated pauses: 477 ms, max pause: 358 ms, pauses over 1000 ms: 0
13:37:05, accumulated pauses: 2049 ms, max pause: 2024 ms, pauses over 1000 ms: 1
13:37:10, accumulated pauses: 4436 ms, max pause: 3818 ms, pauses over 1000 ms: 1
13:37:15, accumulated pauses: 1352 ms, max pause: 1348 ms, pauses over 1000 ms: 1
13:37:45, accumulated pauses: 67159 ms, max pause: 36214 ms, pauses over 1000 ms: 2
13:38:25, accumulated pauses: 5157 ms, max pause: 4882 ms, pauses over 1000 ms: 1
13:38:30, accumulated pauses: 298 ms, max pause: 216 ms, pauses over 1000 ms: 0
13:38:35, accumulated pauses: 33 ms, max pause: 8 ms, pauses over 1000 ms: 0
13:38:40, accumulated pauses: 404 ms, max pause: 382 ms, pauses over 1000 ms: 0
13:38:50, accumulated pauses: 10025 ms, max pause: 10008 ms, pauses over 1000 ms: 1
13:38:55, accumulated pauses: 2580 ms, max pause: 1512 ms, pauses over 1000 ms: 2
13:39:00, accumulated pauses: 25 ms, max pause: 8 ms, pauses over 1000 ms: 0
13:39:05, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:10, accumulated pauses: 26 ms, max pause: 2 ms, pauses over 1000 ms: 0
13:39:15, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:20, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:25, accumulated pauses: 25 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:30, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:35, accumulated pauses: 25 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:39:40, accumulated pauses: 6396 ms, max pause: 6044 ms, pauses over 1000 ms: 1
13:39:45, accumulated pauses: 101 ms, max pause: 94 ms, pauses over 1000 ms: 0
13:39:50, accumulated pauses: 6582 ms, max pause: 6581 ms, pauses over 1000 ms: 1
13:39:55, accumulated pauses: 1010 ms, max pause: 997 ms, pauses over 1000 ms: 0
13:40:00, accumulated pauses: 107 ms, max pause: 19 ms, pauses over 1000 ms: 0
13:40:05, accumulated pauses: 54 ms, max pause: 8 ms, pauses over 1000 ms: 0
13:40:10, accumulated pauses: 169 ms, max pause: 16 ms, pauses over 1000 ms: 0
13:40:15, accumulated pauses: 149 ms, max pause: 30 ms, pauses over 1000 ms: 0
13:40:20, accumulated pauses: 14522 ms, max pause: 14463 ms, pauses over 1000 ms: 1
13:40:35, accumulated pauses: 2011 ms, max pause: 1966 ms, pauses over 1000 ms: 1
13:40:40, accumulated pauses: 60 ms, max pause: 12 ms, pauses over 1000 ms: 0
13:40:45, accumulated pauses: 720 ms, max pause: 700 ms, pauses over 1000 ms: 0
13:40:50, accumulated pauses: 122 ms, max pause: 19 ms, pauses over 1000 ms: 0
13:40:55, accumulated pauses: 78 ms, max pause: 18 ms, pauses over 1000 ms: 0
13:41:00, accumulated pauses: 91 ms, max pause: 15 ms, pauses over 1000 ms: 0
13:41:05, accumulated pauses: 434 ms, max pause: 409 ms, pauses over 1000 ms: 0
13:41:10, accumulated pauses: 44 ms, max pause: 11 ms, pauses over 1000 ms: 0
13:41:15, accumulated pauses: 5681 ms, max pause: 5317 ms, pauses over 1000 ms: 1
13:41:20, accumulated pauses: 5785 ms, max pause: 3847 ms, pauses over 1000 ms: 2
13:41:25, accumulated pauses: 3175 ms, max pause: 3172 ms, pauses over 1000 ms: 1
13:41:30, accumulated pauses: 6892 ms, max pause: 6887 ms, pauses over 1000 ms: 1
13:41:35, accumulated pauses: 4 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:41:40, accumulated pauses: 32 ms, max pause: 2 ms, pauses over 1000 ms: 0
13:41:45, accumulated pauses: 2464 ms, max pause: 1707 ms, pauses over 1000 ms: 1
13:41:50, accumulated pauses: 716 ms, max pause: 658 ms, pauses over 1000 ms: 0
13:41:55, accumulated pauses: 24 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:42:00, accumulated pauses: 23 ms, max pause: 0 ms, pauses over 1000 ms: 0
13:42:05, accumulated pauses: 757 ms, max pause: 706 ms, pauses over 1000 ms: 0
13:42:10, accumulated pauses: 858 ms, max pause: 833 ms, pauses over 1000 ms: 0
13:42:15, accumulated pauses: 30 ms, max pause: 6 ms, pauses over 1000 ms: 0
13:42:20, accumulated pauses: 1132 ms, max pause: 1103 ms, pauses over 1000 ms: 1
13:42:25, accumulated pauses: 688 ms, max pause: 629 ms, pauses over 1000 ms: 0
13:42:30, accumulated pauses: 120 ms, max pause: 15 ms, pauses over 1000 ms: 0
13:42:35, accumulated pauses: 80 ms, max pause: 16 ms, pauses over 1000 ms: 0
13:42:40, accumulated pauses: 1515 ms, max pause: 1496 ms, pauses over 1000 ms: 1
13:42:45, accumulated pauses: 33 ms, max pause: 3 ms, pauses over 1000 ms: 0
13:42:50, accumulated pauses: 663 ms, max pause: 601 ms, pauses over 1000 ms: 0
13:42:55, accumulated pauses: 171 ms, max pause: 22 ms, pauses over 1000 ms: 0
13:43:00, accumulated pauses: 137 ms, max pause: 63 ms, pauses over 1000 ms: 0
13:43:05, accumulated pauses: 1367 ms, max pause: 1288 ms, pauses over 1000 ms: 1
13:43:10, accumulated pauses: 86 ms, max pause: 8 ms, pauses over 1000 ms: 0
13:43:15, accumulated pauses: 250 ms, max pause: 16 ms, pauses over 1000 ms: 0
13:43:20, accumulated pauses: 1003 ms, max pause: 867 ms, pauses over 1000 ms: 0
13:43:25, accumulated pauses: 128 ms, max pause: 83 ms, pauses over 1000 ms: 0
13:43:30, accumulated pauses: 462 ms, max pause: 408 ms, pauses over 1000 ms: 0
13:43:35, accumulated pauses: 851 ms, max pause: 679 ms, pauses over 1000 ms: 0
13:43:40, accumulated pauses: 983 ms, max pause: 927 ms, pauses over 1000 ms: 0
13:43:45, accumulated pauses: 982 ms, max pause: 853 ms, pauses over 1000 ms: 0
13:43:50, accumulated pauses: 1855 ms, max pause: 573 ms, pauses over 1000 ms: 0
13:43:55, accumulated pauses: 1577 ms, max pause: 1159 ms, pauses over 1000 ms: 1
```

**Comment by ufukyilmaz** [comment]
I think this issue is solved with #2529. I just leave it open for tracking.

**Comment by gurbuzali** [comment]
see https://github.com/hazelcast/hazelcast-jet/issues/2409#issuecomment-727892332
